packages/wrangler/src/deploy/deploy.ts

```diff
@@ -220,6 +213,26 @@ export const validateRoutes = (routes: Route[], hasAssets: boolean) => {
 					.join(`\n\n`)
 		);
 	}
+
+	if (mountedAssetRoutes.length > 0 && assets?.directory !== undefined) {
+		const relativeAssetsDir = path.relative(process.cwd(), assets.directory);
+
+		logger.once.warn(
+			`Warning: The following routes will attempt to serve Assets on a mounted path:\n${mountedAssetRoutes
+				.map((route) => {
+					const routeNoScheme = route.replace(/https?:\/\//g, "");
+					const assetPath = path.join(
+						relativeAssetsDir,
+						routeNoScheme.substring(routeNoScheme.indexOf("/"))
+					);
+					return `  • ${route} (Will match assets: ${assetPath})`;
+				})
+				.join("\n")}` +
+				(assets?.routingConfig.has_user_worker
+					? "\n\nRequests not matching an asset will be forwarded to the Worker script."
```

Comment:
We've been given ["your Worker's code" from Brendan](https://github.com/cloudflare/cloudflare-docs/pull/18508#discussion_r1878520026). Let's go with that. I'll update the rest of the docs later this week.

```suggestion
					? "\n\nRequests not matching an asset will be forwarded to your Worker's code."
```